locked-asset/factory/src/locked_asset.rs

```diff
@@ -83,44 +95,60 @@ pub trait LockedAssetModule: token_supply::TokenSupplyModule + token_send::Token
         current_epoch: Epoch,
         old_unlock_milestones: &ManagedVec<UnlockMilestone>,
     ) -> ManagedVec<UnlockMilestone> {
-        let mut new_unlock_milestones = ManagedVec::new();
         let unlock_percent = self.get_unlock_percent(current_epoch, old_unlock_milestones);
         let unlock_percent_remaining = PERCENTAGE_TOTAL - (unlock_percent as u64);
 
         if unlock_percent_remaining == 0 {
-            return new_unlock_milestones;
+            return ManagedVec::new();
         }
 
+        let mut unlock_milestones_merged =
+            ArrayVec::<UnlockMilestone, MAX_MILESTONES_IN_SCHEDULE>::new();
         for old_milestone in old_unlock_milestones.iter() {
             if old_milestone.unlock_epoch > current_epoch {
                 let new_unlock_percent: u64 = (old_milestone.unlock_percent as u64)
                     * PERCENTAGE_TOTAL
                     / unlock_percent_remaining;
-                new_unlock_milestones.push(UnlockMilestone {
+                unlock_milestones_merged.push(UnlockMilestone {
                     unlock_epoch: old_milestone.unlock_epoch,
                     unlock_percent: new_unlock_percent as u8,
                 });
             }
         }
-        let mut sum_of_new_percents = 0u8;
 
-        for new_milestone in new_unlock_milestones.iter() {
-            sum_of_new_percents += new_milestone.unlock_percent;
+        //Compute the leftover percent
+        let mut sum_of_new_percents = 0u8;
+        for milestone in unlock_milestones_merged.iter() {
+            sum_of_new_percents += milestone.unlock_percent;
         }
```

Comment:
Using u8 here for `sum_of_new_percents` could potentially expose you to overflows. I suggest using u16 for the sum.

Reply:
The sum can be 100 max.